tests/test-class-wpseo-frontend.php

```diff
@@ -369,6 +369,12 @@ public function test_robots() {
 		self::$class_instance->options['noindex-author-wpseo'] = true;
 		$this->assertEquals( $expected, self::$class_instance->robots() );
 
+		// test 404
+		$expected = 'noindex,follow';
```

Comment:
I'd advise creating a separate test function for this. When this test fails, it'll be easier to fix a small test than a big test where you will have to debug through a lot of different tests.